common/common.cc

```diff
@@ -109,6 +109,45 @@ bool sorbet::FileOps::removeEmptyDir(const string &path) {
     return true;
 }
 
+void sorbet::FileOps::removeEmptyDirsRecursively(const std::string &dirPath) {
+    DIR *dir;
+    struct dirent *entry;
+    auto dirPathCStr = dirPath.c_str();
+
+    if ((dir = opendir(dirPathCStr)) == nullptr) {
+        switch (errno) {
+            case ENOTDIR: {
+                throw sorbet::FileNotDirException();
+            }
+            default:
+                // Mirrors other FileOps functions: Assume other errors are from FileNotFound.
+                throw sorbet::FileNotFoundException(fmt::format("Couldn't open directory `{}`", dirPath));
+        }
+    }
+
+    while ((entry = readdir(dir)) != nullptr) {
+        if (entry->d_type == DT_DIR) {
+            const auto namelen = strlen(entry->d_name);
+            string_view nameview{entry->d_name, namelen};
+            if (nameview == "."sv || nameview == ".."sv) {
+                continue;
+            }
+
+            auto innerDirPath = fmt::format("{}/{}", dirPath, nameview);
+
+            removeEmptyDirsRecursively(std::move(innerDirPath));
+        } else {
+            throw sorbet::CreateDirException(
```

Comment:
```suggestion
            throw sorbet::RemoveDirException(
```
nit: should we create a separate exception?

Reply:
Looks like this is reused for other FileOps methods, so yes but we should do that separately.

Reply:
On second thought, I'll change it here.

Reply:
fixed